### DISCUSSION TITLE
Datasource config custom http header does not support global variable as value ?

### DISCUSSION THREAD

**Discussion by curvaton** [grafana/grafana-infinity-datasource#411] [Q&A] [answered]
Hi,

I would like to configure the Infinity datasource to pass a global grafana variable as value in the custom http header (e.g the user email address ${__user.email}). That does not seem to work. If I write the user email explicitly that works.

However, when I set the custom http header at the level of the query
![image](https://user-images.githubusercontent.com/40830840/191621374-c4842c17-c301-4935-9bf5-4a6c65f10b91.png)
That way works perfectly. 

Nevertheless I would like to be able to configure this at the level of the datasource configuration instead of having to add a custom header for each Infinity query in each dashboard or panel I manage.

Is that possible ?

Thanks in advance for your help.

**Comment by yesoreyeram** [comment, marked as answer]
Consuming those variables not possible via settings with the current version. Also those variables are expected to work within dashboard scope. Explore, alerting are not supported
